### ISSUE TITLE
Remove Weak ( <TLS 1.2 ) Ciphers from ISLE's Traefik 

### ISSUE THREAD

**Issue by McFateM** [Islandora-Collaboration-Group/ISLE#219]
#### Issue description
This falls somewhere between an issue and a feature request...it's a long story but I will try to be brief.

My ITS security folks have been pushing back server requests lately and not agreeing to make them publicly accessible until they clear whatever vulnerability scan they happen to be running.  That has caused numerous headaches for me since my servers now rely solely on certificates from Let's Encrypt, and those require public access in order to validate.  So, the Catch 22 is that I can't go public until I have a valid cert (and meet other requirements), and I can't get a cert until I go public.   Welcome to my world.  8^(

Perhaps the bigger issue here is that the folks who run the scans and report back don't have a clue what I'm doing so they are unable to help me figure out how to clear these hurdles.  Since it can take up to a day and a half for them to run a new scan, the progress on this issue has been horribly slow and frustrating.  Until yesterday, when I started running my own scans, inside the campus network, so as to be able to test nodes inside the firewall before they go public.  Fortunately, after much gnashing of teeth for the past 2 weeks, I think I have a handle on the issue now.

The specific problem ITS reported was that my server (running some Dockerized apps but not ISLE) was using some "weak" ciphers that were deemed vulnerable.  This was in spite of the fact that my server got an A+ rating from https://www.ssllabs.com. 

Turns out the vulnerabilities are "standard equipment" in Traefik, even the most recent releases.  The TLS 1.0 and 1.1 suites that are in Traefik by default include some 3DES ciphers that are indeed susceptible to at least one exploit.  The fix is relatively easy to make, once you understand what's happening.  

I would love to create a PR with this issue to help others remedy this, but I've made so many other changes to my environment that I'm not sure I can efficiently do so, and I'm also not sure how one would test this on a dev instance of ISLE since those are typically not visible to other hosts on a network (or at least I don't understand how to interrogate them).

#### For an issue, describe steps to reproduce the issue

System setup (OS information, software versions, etc):  Any network accessible instance of ISLE.

Steps:
1.  Run https://www.ssllabs.com/ssltest/analyze.html?d=<target> against your ISLE instance (or anything fronted by Traefik).  In my case that was: https://www.ssllabs.com/ssltest/analyze.html?d=digital.grinnell.edu
2.  Check the output for any "weak" ciphers.  In my case that included the following:
![Screen Shot 2019-04-18 at 1 49 10 PM](https://user-images.githubusercontent.com/8472769/56384110-2eaaeb80-61e1-11e9-8e6f-086c9c31c9a1.png)
3.  Modify your ISLE `./config/proxy/traefik.toml` file to include the following immediately after the line that reads `[entryPoints.https.tls]`.  It should look like this:
```
    [entryPoints.https.tls]
    ##mod## Adding this section to remove default/weak cipher suites per https://docs.traefik.io/configuration/entrypoints/
    minVersion = "VersionTLS12"
    ## Suggested cipher list per https://stackoverflow.com/questions/52128979/recommended-tls-ciphers-for-traefik
    cipherSuites = [
      "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
      "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
      "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
      "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
      "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
      "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
      "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
      "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256"
    ]
    ##end##
```
Pay attention to indentation...it is significant in .toml files like this one!  

4.  Bring your Traefik container down and back up again... 
```
docker stop isle-proxy-ID; docker rm -v isle-proxy-ID; docker compose up -d
```
5.  Repeat the scan from Step 1 above.
6.  Check the new output and look for "weak" ciphers.   There should be NONE left, like this:

![Screen Shot 2019-04-18 at 1 49 34 PM](https://user-images.githubusercontent.com/8472769/56384402-e3450d00-61e1-11e9-852f-f1d65e7ea3a3.png)
  
##### What's the expected result?
See Step 6 above.

##### What's the actual result?
Hopefully NO more weak TLS 1.0 or 1.1 ciphers.

I don't know if these weak ciphers are a real threat or not, but our ITS department seems to think they are a threat, so better to get rid of them than ignore them I suppose.

**Comment by ysuarez** [comment]
I wanted to share a quick way of using the [nmap](https://nmap.org/) CLI tool to check which ciphers and protocols your server is using for https. In this case nmap also provides a "grade" for the strength of the individual ciphers that are being used. I had to use this nmap approach to check on servers on my intranet that the SSLLabs could not check for me since their site could not "see" my intranet servers.

**Here are the nmap params I use:**
nmap --script ssl-enum-ciphers -p 443 library.berklee.edu

**Here is the output:**

Starting Nmap 7.80 ( https://nmap.org ) at 2021-03-10 15:40 EST
Nmap scan report for library.berklee.edu (172.17.2.16)
Host is up (0.00023s latency).
rDNS record for 172.17.2.16: library-prod.berklee.edu

PORT    STATE SERVICE
443/tcp open  https
| ssl-enum-ciphers: 
|   TLSv1.2: 
|     ciphers: 
|       TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (secp256r1) - A
|       TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (dh 2048) - A
|       TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 (dh 2048) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (secp256r1) - A
|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 (dh 2048) - A
|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA (dh 2048) - A
|     compressors: 
|       NULL
|     cipher preference: server
|_  least strength: A

Nmap done: 1 IP address (1 host up) scanned in 1.49 seconds

**Comment by g7morris** [comment]
Closing as "fixed" from ISLE 1.5.5 release & @McFateM @SummittDweller credited . Thanks Mark!
